### DISCUSSION TITLE
Lovely code. Any chance on adding multiple oauth providers?

### DISCUSSION THREAD

**Discussion by thaigertech** [maxcountryman/axum-login#162] [Q&A]
Hi there, 
I much enjoyed your axum login code. 
Wondering if you had any plans on allowing multiple oauth providers.

**Comment by maxcountryman** [comment]
Glad to hear you've found it helpful!

The [multi-auth example](https://github.com/maxcountryman/axum-login/tree/974f78f63ddab6915c2d49896119f992409df192/examples/multi-auth) could be extended to support multiple OAuth providers.  This would work by updating the `Credentials` enum with the desired providers and then specifying callback routes on a per-provider basis.

**Reply by maxcountryman** [reply]
Are you looking for contracting help? Feel free to reach out to me directly.

Generally speaking, the example can be extended such that the config, the routes, and the users provide two or more OAuth providers.

**Reply by thaigertech** [reply]
I did find your details and will reach out
